pkg/virt-controller/watch/vmi.go

```diff
@@ -1223,6 +1198,35 @@ func (c *VMIController) getHotplugVolumes(vmi *virtv1.VirtualMachineInstance, vi
 	return hotplugVolumes
 }
 
+func (c *VMIController) cleanupTriggerPods(virtlauncherPod *k8sv1.Pod, vmi *virtv1.VirtualMachineInstance) error {
+	// Get all pods from the namespace
+	pods, err := c.listPodsFromNamespace(virtlauncherPod.Namespace)
+	if err != nil {
+		return err
+	}
+	zero := int64(0)
+
+	for _, pod := range pods {
+		ownerRef := controller.GetControllerOf(pod)
+		if ownerRef == nil || ownerRef.UID != virtlauncherPod.UID || !isTempPod(pod) {
+			// Skip non trigger pods
+			continue
+		}
+		vmiKey := controller.VirtualMachineKey(vmi)
+		c.podExpectations.ExpectDeletions(vmiKey, []string{controller.PodKey(pod)})
+		err := c.clientset.CoreV1().Pods(pod.GetNamespace()).Delete(pod.Name, &v1.DeleteOptions{
+			GracePeriodSeconds: &zero,
+		})
```

Comment:
doesn't the trigger pod need to be scheduled before being deleted? how do we know the pod was scheduled here? i don't see that check

Reply:
Good point, the pod might not have scheduled yet, let me add that check